CredentialProvider/CredentialProviderPresenter.swift

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import AuthenticationServices
+import SwiftKeychainWrapper
+
+@available(iOS 12, *)
+class CredentialProviderPresenter {
+    weak var view: CredentialProviderViewProtocol?
+    private let profile: Profile
+    
+    init(view: CredentialProviderViewProtocol, profile: Profile = ExtensionProfile(localName: "profile")) {
+        self.view = view
+        self.profile = profile
+    }
+    
+    func extensionConfigurationRequested() {
+        view?.showWelcome()
+        if self.profile.logins.reopenIfClosed() != nil {
+            // At this point there is nothing useful we can do if we cannot open the logins database. Worst case
+            // we skip the synchronization and not all logins will be available to the user if they have changed
+            // since the last time.
+            return
+        }
+        
+        profile.syncCredentialIdentities().upon { result in
+            sleep(2)
```

Comment:
Could we comment why we have a `sleep` here? Is this the only solution? 😬

Reply:
Well, it is a good solution we think because the sync of the credential list usually takes only a few moments and we are doing this when presenting a welcome screen to the user when he enables our app as password manager in settings. If we would not have the delay, the screen would disappear very very quickly and this can be very confusing for the user.